transform/app/tasks/transform/dump/transform.py

```diff
@@ -85,3 +122,42 @@ def transform_data(prev_task_status: Optional[CeleryTaskStatus]) -> CeleryTaskSt
             prev_task_status["reason"] = "Previous task failed"
 
         return prev_task_status
+
+
+def transform_file(
+    file_path: str,
+    transformer: BaseTransformer,
+    spark: SparkSession,
+    s3_client: boto3.client,
+    collection_name: str,
+    input_schema: dict,
+    error_log: dict,
+) -> Optional[DataFrame]:
+    """
+    Transforms the data file using the specified transformer and logs errors if transformation fails.
+
+    Args:
+        file_path (str): Path to the file.
+        transformer (BaseTransformer): Transformer to process the file.
+        spark (SparkSession): Spark session object.
+        s3_client (boto3.client): S3 client for loading data.
+        collection_name (str): Collection name for logging.
+        input_schema (dict): Input schema for loading data.
+        error_log (dict): Dictionary to store transformation errors.
+
+    Returns:
+        Optional[DataFrame]: Transformed DataFrame or None if transformation fails.
+    """
+    try:
+        data = load_file_from_s3(file_path, s3_client)
+        logger.info("Loaded file: %s", file_path)
+        df = load_request_data(spark, data, input_schema, collection_name)
+        df_trans = transformer(spark)(df)
+        logger.info("Transformed file: %s", file_path)
+        return df_trans
+    except Exception as e:
+        tb = traceback.format_exc()
```

Comment:
please remove it before merging. We shouldnt have that on production - it can expose sensitive data. Even though its under VPN. Some day we can manage such a logs when log level == DEBUG

Reply:
Removed